### ISSUE TITLE
Find action seems to do a find, grasp and then again a find

### ISSUE THREAD

**Issue by MaxBaeten** [tue-robotics/action_server#46]
[INFO] [1499788341.827060]: Robot heard 'bring me the coke from the cabinet' {u'actions': [{u'action': u'bring', u'source-location': {u'id': u'cabinet'}, u'object': {u'type': u'coke'}, u'target-location': {u'type': u'person', u'id': u'operator'}}]}
[INFO] [1499788341.827251]: 'I heard bring me the coke from the cabinet, is this correct?'
[INFO] [1499788345.244842]: Question: , grammar: T -> yes | no
[INFO] [1499788345.245202]: Example: 'no'
[INFO] [1499788347.312520]: Robot heard 'yes' None
{"entities":[{"id" : "operator", "type": "waypoint", "pose": { "x": 1.950604, "y": 1.435036, "z": 0.000000, "X": 0.000000, "Y": -0.000000, "Z": 3.056928 }}]}
[INFO] [1499788347.441549]: Starting executing of action PickUp.
[INFO] [1499788347.441988]: Starting executing of action Find.
[INFO] [1499788347.442438]: State machine starting in initial state 'INSPECT_SOURCE_ENTITY' with userdata: 
	[]
[INFO] [1499788347.442797]: State machine starting in initial state 'NAVIGATE_TO_INSPECT' with userdata: 
	[]
[INFO] [1499788347.443232]: State machine starting in initial state 'START_ANALYSIS' with userdata: 
	[]
[INFO] [1499788347.500002]: State machine transitioning 'START_ANALYSIS':'done'-->'NAVIGATE'
[INFO] [1499788347.500544]: State machine starting in initial state 'GET_PLAN' with userdata: 
	[]
[INFO] [1499788347.629294]: choosing best entity from this list (name->weight):
	[('cabinet', 0)]
[INFO] [1499788347.687780]: choosing best entity from this list (name->weight):
	[('cabinet', 0)]
[INFO] [1499788347.770291]: 'Getting there!'
[INFO] [1499788347.795758]: State machine transitioning 'GET_PLAN':'goal_ok'-->'EXECUTE_PLAN'
[INFO] [1499788350.000220]: State machine transitioning 'EXECUTE_PLAN':'blocked'-->'PLAN_BLOCKED'
[INFO] [1499788350.000719]: Plan blocked
[INFO] [1499788350.501870]: Plan free again
[INFO] [1499788350.502292]: State machine transitioning 'PLAN_BLOCKED':'free'-->'EXECUTE_PLAN'
[ERROR] [1499788350.530870]: Received comm state RECALLING when in simple state DONE with SimpleActionClient in NS /amigo/head_ref/action_server
[ERROR] [1499788355.613785]: SimpleActionClient received DONE twice
[INFO] [1499788360.924394]: State machine terminating 'EXECUTE_PLAN':'succeeded':'arrived'
[INFO] [1499788360.924772]: State machine transitioning 'NAVIGATE':'arrived'-->'STOP_ANALYSIS_SUCCEED'
[INFO] [1499788360.991046]: State machine terminating 'STOP_ANALYSIS_SUCCEED':'done':'arrived'
[INFO] [1499788360.991467]: State machine transitioning 'NAVIGATE_TO_INSPECT':'arrived'-->'SEGMENT'
[INFO] [1499788361.025428]: choosing best entity from this list (name->weight):
	[('cabinet', 0)]
Succesfully stored '/home/amigo/ed/kinect/2017-07-11/on_top_of_cabinet/2017-07-11-17-52-43_rgb.png'
[INFO] [1499788364.832861]: >> Segmented 3 objects!
W tensorflow/core/framework/op_def_util.cc:332] Op BatchNormWithGlobalNormalization is deprecated. It will cease to work in GraphDef version 9. Use tf.nn.batch_normalization().
[INFO] [1499788368.559724]:    - Object 0 is a 'tea' (ID: 62372e6c6399279978c335dc65ced894)
[INFO] [1499788368.560197]:    - Object 1 is a 'coke' (ID: acbcb19143a6b4f3cb6013c789652fac)
[INFO] [1499788368.560673]:    - Object 2 is a 'tea' (ID: c597739b64228155db5fe277cddec7b9)
[INFO] [1499788368.561237]: State machine terminating 'SEGMENT':'done':'done'
[INFO] [1499788368.561711]: State machine transitioning 'INSPECT_SOURCE_ENTITY':'done'-->'CHECK_IF_ENTITY_FOUND'
[INFO] [1499788369.199863]: State machine terminating 'CHECK_IF_ENTITY_FOUND':'succeeded':'succeeded'
[INFO] [1499788369.200358]: 'Hey, I found a coke!'
[INFO] [1499788371.061988]: State machine starting in initial state 'PREPARE_GRASP' with userdata: 
	[]
[INFO] [1499788371.075462]: State machine transitioning 'PREPARE_GRASP':'succeeded'-->'NAVIGATE_TO_GRAB'
[INFO] [1499788371.075959]: State machine starting in initial state 'START_ANALYSIS' with userdata: 
	[]
[INFO] [1499788371.190742]: State machine transitioning 'START_ANALYSIS':'done'-->'NAVIGATE'
[INFO] [1499788371.191171]: State machine starting in initial state 'GET_PLAN' with userdata: 
	[]
[INFO] [1499788371.291660]: Navigating to grasp entity id:acbcb19143a6b4f3cb6013c789652fac
[INFO] [1499788371.347327]: 'Getting there!'
[INFO] [1499788371.364575]: State machine transitioning 'GET_PLAN':'goal_ok'-->'EXECUTE_PLAN'
[INFO] [1499788375.974144]: State machine terminating 'EXECUTE_PLAN':'succeeded':'arrived'
[INFO] [1499788375.974737]: State machine transitioning 'NAVIGATE':'arrived'-->'STOP_ANALYSIS_SUCCEED'
[INFO] [1499788376.098432]: State machine terminating 'STOP_ANALYSIS_SUCCEED':'done':'arrived'
[INFO] [1499788376.098886]: State machine transitioning 'NAVIGATE_TO_GRAB':'arrived'-->'GRAB'
Succesfully stored '/home/amigo/ed/kinect/2017-07-11/acbcb19143a6b4f3cb6013c789652fac/2017-07-11-17-52-58_rgb.png'
[INFO] [1499788379.483056]: ID to update: acbcb19143a6b4f3cb6013c789652fac
[INFO] [1499788379.483626]: Updated pose of entity (dx, dy, dz) : (0.000000, 0.000000, 0.000000)
GPD took 0.056222 seconds
[INFO] [1499788379.551608]: Start grasping
[INFO] [1499788390.466994]: Start lift
[INFO] [1499788392.899642]: Start retract
Succesfully stored '/home/amigo/ed/kinect/2017-07-11/acbcb19143a6b4f3cb6013c789652fac/2017-07-11-17-53-14_rgb.png'
[INFO] [1499788396.211368]: State machine terminating 'GRAB':'succeeded':'done'
[INFO] [1499788396.212196]: Starting executing of action Bring.
[INFO] [1499788396.212618]: 'Bring the action!'
[INFO] [1499788397.629180]: Starting executing of action Find.
[INFO] [1499788397.629648]: State machine starting in initial state 'INSPECT_SOURCE_ENTITY' with userdata: 
	[]
[INFO] [1499788397.629944]: State machine starting in initial state 'NAVIGATE_TO_INSPECT' with userdata: 
	[]
[INFO] [1499788397.630296]: State machine starting in initial state 'START_ANALYSIS' with userdata: 
	[]
[INFO] [1499788397.699055]: State machine transitioning 'START_ANALYSIS':'done'-->'NAVIGATE'
[INFO] [1499788397.699700]: State machine starting in initial state 'GET_PLAN' with userdata: 
	[]
[INFO] [1499788397.826250]: choosing best entity from this list (name->weight):
	[('cabinet', 0)]
[INFO] [1499788397.902024]: choosing best entity from this list (name->weight):
	[('cabinet', 0)]
[INFO] [1499788397.993219]: 'I'm on my way!'
[INFO] [1499788398.048400]: State machine transitioning 'GET_PLAN':'goal_ok'-->'EXECUTE_PLAN'
[INFO] [1499788401.054791]: State machine terminating 'EXECUTE_PLAN':'succeeded':'arrived'
[INFO] [1499788401.055418]: State machine transitioning 'NAVIGATE':'arrived'-->'STOP_ANALYSIS_SUCCEED'
[INFO] [1499788401.135759]: State machine terminating 'STOP_ANALYSIS_SUCCEED':'done':'arrived'
[INFO] [1499788401.136148]: State machine transitioning 'NAVIGATE_TO_INSPECT':'arrived'-->'SEGMENT'
[INFO] [1499788401.163588]: choosing best entity from this list (name->weight):
	[('cabinet', 0)]
Succesfully stored '/home/amigo/ed/kinect/2017-07-11/on_top_of_cabinet/2017-07-11-17-53-22_rgb.png'
[INFO] [1499788403.723626]: >> Segmented 2 objects!
[INFO] [1499788406.130146]:    - Object 0 is a 'tea' (ID: 181c5116f1ef8a87bab279e4469d972a)
[INFO] [1499788406.130531]:    - Object 1 is a 'tea' (ID: c597739b64228155db5fe277cddec7b9)
[INFO] [1499788406.131019]: State machine terminating 'SEGMENT':'done':'done'
[INFO] [1499788406.131396]: State machine transitioning 'INSPECT_SOURCE_ENTITY':'done'-->'CHECK_IF_ENTITY_FOUND'
[INFO] [1499788406.682588]: State machine terminating 'CHECK_IF_ENTITY_FOUND':'failed':'not_found'
[INFO] [1499788406.682989]: 'I don't see what I am looking for here.'

[INFO] [1499788409.148099]: 'Moving to the meeting point.'
[INFO] [1499788409.174660]: State machine starting in initial state 'START_ANALYSIS' with userdata: 
	[]
[INFO] [1499788409.240651]: State machine transitioning 'START_ANALYSIS':'done'-->'NAVIGATE'
[INFO] [1499788409.240815]: State machine starting in initial state 'GET_PLAN' with userdata: 
	[]
[INFO] [1499788409.400305]: 'I will be there in a sec!'
[INFO] [1499788409.420043]: State machine transitioning 'GET_PLAN':'goal_ok'-->'EXECUTE_PLAN'
[INFO] [1499788428.355434]: State machine terminating 'EXECUTE_PLAN':'succeeded':'arrived'
[INFO] [1499788428.355578]: State machine transitioning 'NAVIGATE':'arrived'-->'STOP_ANALYSIS_SUCCEED'
[INFO] [1499788428.443135]: State machine terminating 'STOP_ANALYSIS_SUCCEED':'done':'arrived'
[INFO] [1499788428.443784]: ' I found a coke.  I picked up the coke.  I found a coke.  I picked up the coke.  I was unable to find the coke.  I performed 1 task so far, still going strong!'

**Comment by rokusottervanger** [comment]
This happened because the action queue was not cleared after an action failed. Fixed now
